### DISCUSSION TITLE
[2.x] Allow using anonymous Blade component for layout

### DISCUSSION THREAD

**Discussion by redgluten** [livewire/livewire#1565] [4. General]
From the docs: 

> By default in 1.x, Livewire renders your page-level components using a traditional Blade layout located in resources/layouts/app.blade.php. In 2.0, Livewire uses the same layout file as a default, however, it now expects you are using the new Blade component $slot syntax in the layout. [...]

All fine and good, components FTW, but that means you need to create a class-based Blade component (`app\View\Component\Layout`) to declare the view since by default Laravel only picks up anonymous components located under `resources/components`. Or am I missing something obvious?

If not, well, that kinda sucks: I have no need for a class-based layout component. So I have a very naive ready-to-PR [change](https://github.com/redgluten/livewire/commit/7eae0f1b202f2599f17b7dc4ef04243c2c21007f) that will make Livewire search first for a `components.layout` (inspired by the [Laracasts video](https://laracasts.com/series/blade-component-cookbook/episodes/1) on the component-as-layout trick) and then fallback to the `layouts.app` if needed. I’d gladly submit it if that indeed makes sense.

**Comment by mikebronner** [comment]
Possibly related to https://github.com/livewire/livewire/discussions/1730? I'm having problems getting Livewire to work in apps the use layout-components. I tried the solution in your PR, but it ended up not working either. Not sure what the solution here might be.
